src/main/java/org/opentripplanner/transit/service/TransitModel.java

```diff
@@ -315,6 +316,25 @@ public void addFeedInfo(FeedInfo info) {
     this.feedInfoForId.put(info.getId(), info);
   }
 
+  /**
+   * Returns the time zone for the transit model. This is used to interpret times in API requests.
+   * Ideally we would want to interpret times in the time zone of the geographic location where the
+   * origin/destination vertex or board/alight event is located. This may become necessary when we
+   * start making graphs with long distance train, boat, or air services.
+   */
+  public ZoneId getTimeZone() {
+    return timeZone;
+  }
+
+  public void setTimeZone(ZoneId timeZone) {
+    if (this.timeZone != null) {
+      throw new IllegalStateException("Timezone can't be re-set");
+    } else if (timeZone != null) {
+      this.timeZone = timeZone;
+      this.timeZoneExplicitlySet = true;
+    }
+  }
```

Comment:
Tip!
```suggestion
  public void initTimeZone(ZoneId timeZone) {
    if (this.timeZone != null) {
      throw new IllegalStateException("Timezone can't be re-set");
    } 
    if(timeZone == null) {
      return;
    }
    this.timeZone = timeZone;
    this.timeZoneExplicitlySet = true;
  }
```
Ignore `timeZone is null` and `this.timeZone != null` are preconditions - witch is clearer from the suggested code. The original code is within the "developer license", but the suggestion is a bit more readable. Using `init` and `set` as a prefix indicate that you should call this early and only once.

Reply:
Yes, this is the way I usually structure the code, much cleaner this way. Also the suggested name is much cleaner, and makes you understand what it does.